tests/test_agents.py

```diff
@@ -0,0 +1,648 @@
+from __future__ import annotations
+
+import itertools
+import json
+import logging
+import re
+from pathlib import Path
+from typing import Any, cast
+from unittest.mock import patch
+
+import anyio
+import pytest
+from pydantic import ValidationError
+from pytest_subtests import SubTests
+
+import paperqa
+from paperqa.docs import Docs
+from paperqa.llms import LangchainLLMModel
+from paperqa.types import Answer, Context, Doc, PromptCollection, Text
+from paperqa.utils import get_year
+
+SKIP_AGENT_TESTS = False
+
+try:
+    from langchain.agents.openai_functions_agent.base import OpenAIFunctionsAgent
+    from langchain_openai import ChatOpenAI
+    from tenacity import Retrying, retry_if_exception_type, stop_after_attempt
+
+    from paperqa.agents import aagent_query
+    from paperqa.agents.docs import (
+        compute_cost,
+        stream_answer,
+        stream_evidence,
+        stream_filter,
+        update_doc_models,
+    )
+    from paperqa.agents.models import (
+        AgentPromptCollection,
+        AgentStatus,
+        AnswerResponse,
+        MismatchedModelsError,
+        QueryRequest,
+    )
+    from paperqa.agents.prompts import STATIC_PROMPTS
+    from paperqa.agents.search import get_directory_index
+    from paperqa.agents.tools import (
+        GatherEvidenceTool,
+        GenerateAnswerTool,
+        PaperSearchTool,
+        SharedToolState,
+    )
+except ImportError:
+    SKIP_AGENT_TESTS = True
+
+if SKIP_AGENT_TESTS:
+    pytest.skip("agents module is not installed", allow_module_level=True)
+
+PAPER_DIRECTORY = str(Path(__file__).parent)
```

Comment:
```suggestion
PAPER_DIRECTORY = Path(__file__).parent
```

Just a few lines below this it seems we use `anyio.Path`. Let's just stick with `Path` the entire time